utils.js

```diff
@@ -45,15 +45,15 @@ async function checkForPlaystationDirectRedirect(checkInterval, onSuccess, versi
     const response = await page.goto(url);
     const responseBody = await response.text();
     const responseStatus = await response.status();
-
+    const cookies = await page.cookies();
     await context.close();
 
     // Uncomment to see the response body for debugging
     // console.log(`Response body: ${responseBody}`);
     // console.log(`Response status: ${responseStatus}`);
 
-    if (responseBody.indexOf("queue-it_log") > 0 && 
-        responseBody.indexOf("softblock") === -1) {
+    if ((responseBody.indexOf("queue-it_log") > 0 && 
+        responseBody.indexOf("softblock") === -1) || (cookies.length && cookies.some(cookie => cookie.name === 'Queue-it'))) {
```

Comment:
@alexsheehan Can you tell me more about this cookie? Any idea when/what circumstances Sony creates it?

Reply:
Hmm looks like it's created when the user is directed to the softblock page here: https://direct-queue.playstation.com/softblock/?c=sonyied&e=psdirectprodku1&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-digital-edition-console.3005817&cid=en-US

Reply:
From looking at the requests before/after the queue starts, I think this cookie only gets set when you join the real queue. If you go to the softblock page, it makes a cookie called `QueueITAccepted-SDFrts345E-V3_psdirectprodku1` but from what I can tell, the 'Queue-it` cookie only gets set when there's an actual queue. Here are responses before/after queue:
```
// No redirect detected. Trying again...
// Number of tries 150

HTTPResponse {
  _contentPromise: null,
  _headers: {
    date: 'Fri, 11 Dec 2020 21:21:35 GMT',
    'content-type': 'text/html; charset=utf-8',
    'transfer-encoding': 'chunked',
    connection: 'keep-alive',
    server: 'Kestrel',
    p3p: 'CP="NOI ADM DEV PSAi COM NAV OUR OTR STP IND DEM"'
  },
  _client: CDPSession {
    eventsMap: Map { ... },
    emitter: { ... },
    _callbacks: Map {},
    _connection: Connection { ... },
    _targetType: 'page',
    _sessionId: 'B829BC928C3AA3EA4520D221F8C63416'
  },
  _request: HTTPRequest {
    _failureText: null,
    _response: [Circular],
    _fromMemoryCache: false,
    _interceptionHandled: false,
    _headers: {
      'upgrade-insecure-requests': '1',
      'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/88.0.4298.0 Safari/537.36'
    },
    _client: CDPSession { ... },
    _requestId: 'AC5BEC27B1337C7594179B2E8C43A3C8',
    _isNavigationRequest: true,
    _interceptionId: null,
    _allowInterception: false,
    _url: 'https://direct-queue.playstation.com/softblock/?c=sonyied&e=psdirectprodku1&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-console.3005816&cid=en-US',
    _resourceType: 'document',
    _method: 'GET',
    _postData: undefined,
    _frame: Frame {
      _url: 'https://direct-queue.playstation.com/softblock/?c=sonyied&e=psdirectprodku1&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-console.3005816&cid=en-US',
      _detached: false,
      _loaderId: 'AC5BEC27B1337C7594179B2E8C43A3C8',
      _lifecycleEvents: [Set],
      _frameManager: [FrameManager],
      _parentFrame: null,
      _id: '477F76C30EDEBFE4862D3C81E427AF03',
      _mainWorld: [DOMWorld],
      _secondaryWorld: [DOMWorld],
      _childFrames: [Set],
      _name: undefined
    },
    _redirectChain: [ [HTTPRequest], [HTTPRequest] ]
  },
  _bodyLoadedPromiseFulfill: [Function],
  _bodyLoadedPromise: Promise { null },
  _remoteAddress: { ip: '54.201.159.129', port: 443 },
  _status: 200,
  _statusText: 'OK',
  _url: 'https://direct-queue.playstation.com/softblock/?c=sonyied&e=psdirectprodku1&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-console.3005816&cid=en-US',
  _fromDiskCache: false,
  _fromServiceWorker: false,
  _securityDetails: SecurityDetails { ... }
}
// No redirect detected. Trying again...
// Number of tries 151

HTTPResponse {
  _contentPromise: null,
  _headers: {
    date: 'Fri, 11 Dec 2020 21:21:44 GMT',
    'content-type': 'text/html; charset=utf-8',
    'transfer-encoding': 'chunked',
    connection: 'keep-alive',
    server: 'Kestrel',
    'cache-control': 'no-store,no-cache',
    pragma: 'no-cache',
    'set-cookie': 'Queue-it=u=3a18ac13-6c2a-47ef-85c9-ec7424b254e8; expires=Sat, 11 Dec 2021 21:21:44 GMT; path=/; secure; httponly; samesite=none',
    p3p: 'CP="NOI ADM DEV PSAi COM NAV OUR OTR STP IND DEM"'
  },
  _client: CDPSession {
    eventsMap: Map { ... },
    emitter: { ... },
    _callbacks: Map {},
    _connection: Connection { ... },
    _targetType: 'page',
    _sessionId: 'E53F04E378282D66B09BC3DDA4B7B4E1'
  },
  _request: HTTPRequest {
    _failureText: null,
    _response: [Circular],
    _fromMemoryCache: false,
    _interceptionHandled: false,
    _headers: {
      'upgrade-insecure-requests': '1',
      'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/88.0.4298.0 Safari/537.36'
    },
    _client: CDPSession { ... },
    _requestId: '512ADF9FE0FE57236E2C8A1650F131B9',
    _isNavigationRequest: true,
    _interceptionId: null,
    _allowInterception: false,
    _url: 'https://direct-queue.playstation.com/?c=sonyied&e=psdirectprodku1&ver=v3-javascript-3.6.3&cver=144&man=PS%20Direct%20Prod%20KU%20Action&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-console.3005816&kupver=akamai-1.0.2',
    _resourceType: 'document',
    _method: 'GET',
    _postData: undefined,
    _frame: Frame {
      _url: 'https://direct-queue.playstation.com/?c=sonyied&e=psdirectprodku1&ver=v3-javascript-3.6.3&cver=144&man=PS%20Direct%20Prod%20KU%20Action&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-console.3005816&kupver=akamai-1.0.2',
      _detached: false,
      _loaderId: '512ADF9FE0FE57236E2C8A1650F131B9',
      _lifecycleEvents: [Set],
      _frameManager: [FrameManager],
      _parentFrame: null,
      _id: '5C8916795D04CED0D2FDF06F4FDC8055',
      _mainWorld: [DOMWorld],
      _secondaryWorld: [DOMWorld],
      _childFrames: [Set],
      _name: undefined
    },
    _redirectChain: [ [HTTPRequest] ]
  },
  _bodyLoadedPromiseFulfill: [Function],
  _bodyLoadedPromise: Promise { null },
  _remoteAddress: { ip: '54.201.159.129', port: 443 },
  _status: 200,
  _statusText: 'OK',
  _url: 'https://direct-queue.playstation.com/?c=sonyied&e=psdirectprodku1&ver=v3-javascript-3.6.3&cver=144&man=PS%20Direct%20Prod%20KU%20Action&t=https%3A%2F%2Fdirect.playstation.com%2Fen-us%2Fconsoles%2Fconsole%2Fplaystation5-console.3005816&kupver=akamai-1.0.2',
  _fromDiskCache: false,
  _fromServiceWorker: false,
  _securityDetails: SecurityDetails { ... }
}
// Found it! Opening queue now...
```